database.go

```diff
@@ -94,9 +98,10 @@ func (sd *stateDatabase) ContractCodeSize(_, codeHash common.Hash) (int, error)
 // StateAccount satisfies Database, it returns the types.StateAccount for a provided address and block hash
 func (sd *stateDatabase) StateAccount(addressHash, blockHash common.Hash) (*types.StateAccount, error) {
 	res := StateAccountResult{}
-	err := sd.pgdb.QueryRow(context.Background(), GetStateAccount, addressHash.Hex(), blockHash.Hex()).Scan(&res)
+	err := sd.pgdb.QueryRow(context.Background(), GetStateAccount, addressHash.Hex(), blockHash.Hex()).
```

Comment:
Is this a `pgx` particularity? With `sqlx` IIRC you can unpack into the entire struct (as long as it has the `db:""` tagged public fields).

Reply:
Right, that only works in sqlx - see https://github.com/jackc/pgx/issues/760
In pgx v5, there is [RowToStructByName](https://pkg.go.dev/github.com/jackc/pgx/v5#RowToStructByName), but it requires Go 1.19